[Hwww23] iP

Hwww23 opened the pull request:
## **PepegPro**
> “Your mind is for having ideas, not holding them.” – David Allen (source)

DukePro frees your mind from having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/Hwww23/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks 
- [ ] Managing deadlines (coming soon) 
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Hwww23 committed:
add level 0

Hwww23 committed:
add level 0

Hwww23 committed:
Merge remote-tracking branch 'origin/master'

Hwww23 committed:
add level 1

Hwww23 committed:
no message

Hwww23 committed:
add level 2

Hwww23 committed:
add level 3

Hwww23 committed:
add level 4

Hwww23 committed:
testing actual vs expected

Hwww23 committed:
add level 5

Hwww23 committed:
add level 6

Hwww23 committed:
add enum

Hwww23 committed:
Add duke.txt
Update Duke.java
Update Event.java
Update Todo.java

Hwww23 committed:
Merge branch 'branch-Level-7'

Javiery3889 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,175 @@
+import java.io.*;
+import java.util.*;
```
Following our coding standard, imported classes should be listed explicitly. For example,
```suggestion
import java.io.BufferedReader;
```
I noticed the same issue in several other places as well.

Javiery3889 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,175 @@
+                    String s, s1;
+                    Task task;
+
+                    switch (type) {
+                        case "T":
+                            s = loadInput[2].trim();
+                            task = new Todo(s);
+
+                            if (done.equals("1")) {
+                                task.markAsDone();
+                            }
+
+                            myList.addItem(task);
+                            break;
+                        case "D":
+                            s = loadInput[2].trim();
+                            s1 = loadInput[3].trim();
+                            task = new Deadline(s, s1);
+
+                            if (done.equals("1")) {
+                                task.markAsDone();
+                            }
+
+                            myList.addItem(task);
+                            break;
+                        case "E":
+                            s = loadInput[2].trim();
+                            s1 = loadInput[3].trim();
+                            String s2 = loadInput[4].trim();
+                            task = new Event(s, s1, s2);
+
+                            if (done.equals("1")) {
+                                task.markAsDone();
+                            }
+
+                            myList.addItem(task);
+                            break;
+                    }
+                }
```
Following our coding standards, `case` clauses **should not** have indentation in switch statements.

Javiery3889 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+        return this.description;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
```
Should the logic in this method be separated instead? Perhaps you can consider using a getter method for the boolean variable `isDone` and format the output based on the variable in the `toString()` method.

Javiery3889 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+public class Event extends Task {
+    protected String start;
+    protected String end;
```
Any reason why you decided to set the access modifier of these two variables to `protected` instead of `private`?

Javiery3889 commented on src/main/java/MyList.java:
```diff
@@ -0,0 +1,44 @@
+import java.util.*;
+public class MyList {
+    private List<Task> items;
+
+    public MyList() {
+        this.items = new ArrayList<>();
+    }
+
+    public String addItem(Task t) {
+        this.items.add(t);
+        return "Got it. I've added this task:\n" + t.toString() + "\nNow you have " + this.items.size() + " tasks in the list.";
+    }
+
+    public String getItems() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Here are the tasks in your list:\n");
+        int index = 1;
+
+        for (Task task : this.items) {
+            String itemString = String.format("%d. %s\n", index, task.toString());
+            stringBuilder.append(itemString);
+            index++;
+        }
+
+        return stringBuilder.toString();
+    }
+
+    public String markTask(int index) {
+        if (index < 1 || index > this.items.size()) {
+            return "Number is outside length of list";
+        } else {
+            return "Nice! I've marked this task as done:\n" + this.items.get(index - 1).markAsDone();
+        }
+    }
```
I like your naming of methods adhere to the coding standard 👍. One small nit would be perhaps to choose between `task` or `item` in your method names in this class

Javiery3889 reviewed (commented):
Overall, I found your code easy to follow for the most part except for logic in `Duke.java` as it is quite long. You can consider splitting the class into smaller classes if that makes sense. I noted a few minor coding standard violations too. Press on! 👍

Hwww23 committed:
Update Duke.java
Update Deadline.java
Update Event.java
Update duke.txt

Hwww23 committed:
Merge branch 'branch-Level-8'

Hwww23 committed:
Add Ui.java, Storage.java and Parser.java
Update Duke.java, MyList.java, Task.java, Todo.java, Deadline.java, Event.java

Hwww23 committed:
no message

Hwww23 committed:
Add checkstyle.xml, suppressions.xml
Update build.gradle

Hwww23 committed:
Add duketest.txt, ParserTest.java and StorageTest.java

ivan24004 commented on src/main/java/duke/MyList.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    public MyList(List<Task> t) {
+        this.items = t;
```
I think `this` can be omitted from the line

ivan24004 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,135 @@
+        Request request = getRequest(userInput);
+
+        switch (request) {
+            case BYE:
```
I think this should be deindented to follow our switch case convention.

ivan24004 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,31 @@
+    public String toString() {
+        DateTimeFormatter customFormatter = DateTimeFormatter.ofPattern("MMM d yyyy h:mma");
+        String formattedDateTime = this.start.format(customFormatter);
+        String formattedDateTime1 = this.end.format(customFormatter);
```
Perhaps change `formattedDateTime1` to something more descriptive, like `formattedEndDateTime`

ivan24004 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,93 @@
+            String taskString, byString, dateTimePattern, fromString, toString;
+            Task task;
+
+            switch (type) {
```
Similar comment as above regarding switch-case indentation.

ivan24004 commented on src/test/java/duke/StorageTest.java:
```diff
@@ -0,0 +1,40 @@
+public class StorageTest {
+
+    @Test
+    public void testSaveAndLoad() throws DukeException, IOException {
```
I don't think this follows our coding standard for naming tests, it should follow the format: `featureUnderTest_testScenario_expectedBehavior()`

ivan24004 commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,21 @@
+public class ParserTest {
+
+    @Test
+    public void testGetRequest() {
```
I don't think this follows our coding standard for naming tests, it should follow the format: `featureUnderTest_testScenario_expectedBehavior()`

LWS49 commented on src/main/java/duke/MyList.java:
```diff
@@ -0,0 +1,54 @@
+
+    public String markTask(int index) {
+        if (index < 1 || index > this.items.size()) {
+            return "Number is outside length of list";
```
Would it have been better to throw an exception instead, since this is an error in user input?

LWS49 commented on src/main/java/duke/MyList.java:
```diff
@@ -0,0 +1,54 @@
+
+    public String delete(int index) {
+        if (index < 1 || index > this.items.size()) {
+            return "Number is outside length of list";
```
Same for here

Hwww23 committed:
Add Javadoc comments for code documentation

Hwww23 committed:
Update Deadline.java, Event.java, MyList.java, Parser.java, Storage.java, ParserTest.java and StorageTest.java

To adhere to java coding standards.

Hwww23 committed:
Update Mylist.java, Parser.java, Task.java

To implement find function

Hwww23 committed:
Merge branch 'branch-A-CodingStandard'

Hwww23 committed:
Resolve merge conflicts in Parser.java

Hwww23 committed:
Update Mylist.java, ParserTest.java, StorageTest.java and build.gradle

Hwww23 committed:
Added GUI

Hwww23 committed:
Merge branch 'branch-Level-10'

Hwww23 committed:
Update README.md

Hwww23 committed:
Update README.md

Hwww23 committed:
Update README.md

Hwww23 committed:
Update duke.txt

Hwww23 committed:
Update Duke.java and Storage.java

To resolve a bug with saving task

Hwww23 committed:
Update MyList.java, Storage.java

Added assertions to addTask method in MyList.java and save
method in Storage.java.

- Ensure that task added into the list should not be null or have
an empty description.
- Ensure that the list being saved into duke.txt is not null.

Hwww23 committed:
Merge pull request #2 from Hwww23/branch-A-Assertions

Update MyList.java, Storage.java

Hwww23 committed:
Add AddCommand.java, ByeCommand.java, Command.java,
DeleteCommand.java, FindCommand.java, InvalidCommand.java,
ListCommand.java and Mark.java and update Ui.java, Storage.java,
Parser.java and Duke.java

There were methods that were too long which caused deep nesting
in Storage.java and Parser.java.

The Command class was added to reduce the length of methods by
making parser return a specific child class of the Command class,
the required operation is then executed depending on the command
class.

Hwww23 committed:
Update Storage.java

Conflicts for different method parameter names.

Hwww23 committed:
Merge pull request #3 from Hwww23/branch-A-CodeQuality

Add Command class

Hwww23 committed:
Update README.md

Hwww23 committed:
Update README.md

Hwww23 committed:
Add RecurringTask.java, TaskScheduler.java, RecurCommand.java and
update Parser.java, MyList.java, Event.java and Deadline.java

A requirement to provide support for recurring tasks.

Provide a RecurringTask interface for deadlines and events and
and implement a scheduler so date time can update automatically

Hwww23 committed:
Update README.md

Hwww23 committed:
Update README.md

Hwww23 committed:
Add Ui.png

Hwww23 committed:
Update Ui.java and Main.java

Rename chatbot name

Hwww23 committed:
Update README.md

Hwww23 committed:
Update README.md

Update user guide

Hwww23 committed:
Update build.gradle

update name of jar file

Hwww23 committed:
Update README.md

Hwww23 committed:
Update README.md, Ui.png, duke.txt, Deadline.java, Event.java and
Storage.java

Whether a task was recurring a not was not saved and loaded
properly.

Implement conditions in Storage.java to load recurring task
correctly and made some changes to the save() method in
Deadline.java and Event.java for proper saving.

Hwww23 committed:
Fix jar

Hwww23 committed:
Fix jar

Hwww23 committed:
Update Parser.java and Duke.java

Fix invalid datetime bug, e.g. 2022-03-33 2500 and invalid
userinput, e.g. todoabc
